### PR TITLE
parse dnadiff output

### PR DESCRIPTION
**Pull request by kiepczi**
This would close issue #17. 

This PR adds:
- Parsing of `.mcoords` and `.qdiff` files to replicate values reported by `dnadiff`.
- Script used to generate expected matrices from `.report` files. The matrices could not be generated using legacy `pyANI`, as this method is new for `pyani-plus` and has not been available in legacy `pyANI`.
- Pytest tests to check if the reported values are consistent with the output from `dnadiff`.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [X] This change requires a documentation update
- [ ] This is a documentation update

## Action Checklist

- [X] Work on a single issue/concept (if there are multiple separate issues to address, please use a separate pull request for each)
- [X] Fork the `pyani-plus` repository under your own account (please [allow write access for repository maintainers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
- [X] Set up an appropriate development environment (please see `CONTRIBUTING.md` for this repository)
- [X] Create a new branch with a short, descriptive name
- [X] Work on this branch
  - [X] style guidelines have been followed (please see `CONTRIBUTING.md` for this repository)
  - [X] new code is commented, especially in hard-to-understand areas
  - [ ] corresponding changes to documentation have been made
  - [X] tests for the change have been added that demonstrate the fix or feature works
- [X] Test locally with `pytest-plus -v` **non-passing code will not be merged**
- [] Rebase against `origin/master`
- [ ] Check changes with linting/formatting tools before submission (please see `CONTRIBUTING.md` for this repository)
- [X] Commit branch
- [X] Submit pull request, describing the content and intent of the pull request
- [X] Request a code review
- [X] Continue the discussion at [`Pull requests` section](https://github.com/widdowquinn/pyani-plus/pulls) in the `pyan-plus` repository

### CODE REVIEW

File: pyani_plus/methods/method_dnadiff.py
Comment (widdowquinn) [inline pyani_plus/methods/method_dnadiff.py:26]: Duplicate licenses - can we cut this down before merging?

**Review by widdowquinn — COMMENTED** [review]
Looks good - thank you @kiepczi !

**Comment by widdowquinn** [comment]
We justify the `.mcoords` based calculations here, in the online design docs: https://pyani-plus.github.io/design-documentation/deltafiles.html

File: pyani_plus/methods/method_dnadiff.py
Comment (peterjc) [reply pyani_plus/methods/method_dnadiff.py:26]: As discussed, this was "2024-present" versus the expected "2024" difference meaning the license check failed, so the "correct" license was inserted above. See #77 for more, but the year-only style allows automatic updates to the current year.

We might want to explore the fuzzy matching support in the license checking rule...

**Comment by kiepczi** [comment]
Hi @peterjc! 

I made the changes as suggested at yesterday's meeting. I would appreciate if you could give me a green light 🚥 to merge this into main :)

**Comment by peterjc** [comment]
I was wondering why ``tests/generate_fixtures/generate_target_dnadiff_matrices.py`` was calling pyani-plus code, but this is a new method with no reference implementation in legacy pyani

File: tests/conftest.py
Comment (peterjc) [inline tests/conftest.py:215]: Do you actually use the second feature ``dnadiff_matrices``?

Comment (kiepczi) [reply tests/conftest.py:215]: Good spot! I was planning to use it, but I switched to an alternative! I will remove this used fixture.

**Review by peterjc — APPROVED** [review]
Approved with unused feature ``dnadiff_matrices`` removed, but I would prefer not to do a standard merge as it will leave a complicated history. See other comment suggesting squash-and-merge, or that I try a rebase.

**Comment by kiepczi** [comment]
> I was wondering why `tests/generate_fixtures/generate_target_dnadiff_matrices.py` was calling pyani-plus code, but this is a new method with no reference implementation in legacy pyani

I needed to reuse one of the functions from `method_anim`. This part of the code is likely to be removed once I implement the self-to-self comparisons, but I have decided to leave it for another branch.

**Comment by kiepczi** [comment]
> Approved with unused feature `dnadiff_matrices` removed, but I would prefer not to do a standard merge as it will leave a complicated history. See other comment suggesting squash-and-merge, or that I try a rebase.

If you want to do this, I will be more than happy. Otherwise, I would probably end up complicating the history even more.

**Comment by peterjc** [comment]
About to have lunch here - I'll do it after that (go ahead with removing the second fixture now if you like).

**Comment by peterjc** [comment]
In this case if we restrict ourselves to just your commits, and drop mine, and interactive rebase ought to be easy as we didn't touch the same files.

There is a configuration option to change the simple text you get via ``git rebase main -i`` (interactive) which uses the one line commit messages by default, see https://stackoverflow.com/questions/35851671/is-there-a-way-to-list-the-commits-author-in-git-rebase-i-interactive

As a one off config change:

```
$ git -c "rebase.instructionFormat=(%an <%ae>) %s" rebase -i 
```

I could then easily remove (or change ``pick ...`` to ``drop ...``) all the commits by me, keeping just those by you and the pre-commit bot. It turned out the second pre-commit commit was from the main branch so I used ``git rebase --skip`` on that.

I then confirmed ``make test`` was fine, and did ``git push origin issue_17_calc --force`` (the rare case where we do use a forced push as we want to rewrite the pull request history).

--

Generally while working on a branch I would avoid doing a merge from main *unless* you are knowingly working on files which have changed on main and there is a conflict. I find this keeps the history cleaner and I find it easier to keep track of what I am doing.

**Comment by peterjc** [comment]
You can go ahead and merge this if there is nothing amiss after the rebase

**Comment by peterjc** [comment]
Note if you said "closes #17" rather than "closes issue #17", then GitHub would have auto-closed it when the pull request was merged.

There are other variants too, like "fixes #17", see:

https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Pull request merged**
